src/Owin.Security.OpenIdConnect.Server/OpenIdConnectServerHandler.Endpoints.cs

```diff
@@ -1625,97 +1649,235 @@ internal partial class OpenIdConnectServerHandler : AuthenticationHandler<OpenId
-            }
+                var endpointContext = new ValidationEndpointContext(Context, Options, request, ticket);
 
-            // Client applications and resource servers are strongly encouraged
-            // to provide an audience parameter to mitigate confused deputy attacks.
-            // See http://en.wikipedia.org/wiki/Confused_deputy_problem.
-            var audiences = ticket.Properties.GetAudiences();
-            if (audiences.Any() && !audiences.ContainsSet(request.GetAudiences())) {
-                await SendErrorPayloadAsync(new OpenIdConnectMessage {
-                    Error = OpenIdConnectConstants.Errors.InvalidGrant,
-                    ErrorDescription = "Invalid access token received: " +
-                        "the audience doesn't correspond to the registered value"
-                });
+                // Add the claims extracted from the access token.
+                foreach (var claim in ticket.Identity.Claims)
+                {
+                    endpointContext.Claims.Add(claim);
+                }
 
-                return;
-            }
+                await Options.Provider.ValidationEndpoint(endpointContext);
 
-            var notification = new ValidationEndpointContext(Context, Options, request, ticket);
+                // Flow the changes made to the authentication ticket.
+                ticket = endpointContext.AuthenticationTicket;
 
-            // Add the claims extracted from the access token.
-            foreach (var claim in ticket.Identity.Claims) {
-                notification.Claims.Add(claim);
-            }
+                if (endpointContext.HandledResponse)
+                {
+                    return;
+                }
 
-            await Options.Provider.ValidationEndpoint(notification);
+                if (ticket.Properties.ExpiresUtc.HasValue && ticket.Properties.ExpiresUtc < Options.SystemClock.UtcNow)
+                {
+                    Options.Logger.WriteVerbose("expired token");
+                    payload.Add("active", false);
+                }
 
-            // Flow the changes made to the authentication ticket.
-            ticket = notification.AuthenticationTicket;
+                //TODO: do more testing for token validity.
 
-            if (notification.HandledResponse) {
-                return;
-            }
+                else
+                {
+                    payload.Add(OpenIdConnectConstants.Claims.Active, true);
 
-            var payload = new JObject();
+                    var scopes = ticket.GetScopes();
+                    if (scopes.Any())
+                    {
+                        payload.Add(OpenIdConnectConstants.Claims.Scope, string.Join(" ", scopes));
+                    }
+                    
+                    payload.Add(OpenIdConnectConstants.Claims.ClientId, request.ClientId);
+
+                    // TODO: Re-examine whether ticket.Identity will always be the resource owner who authorized the token
```

Comment:
It may also represent an application (when using the client credentials grant).
`ClaimsIdentity.Name` is not guaranteed to be non-null here.